subprojects/model-core/src/main/java/org/gradle/api/internal/provider/ValidatingMapEntryCollector.java

```diff
@@ -34,15 +36,26 @@ public ValidatingMapEntryCollector(Class<K> keyType, Class<V> valueType, ValueSa
 
     @Override
     public void add(K key, V value, Map<K, V> dest) {
+        Preconditions.checkNotNull(
+            key,
+            "Cannot get the value of a property of type %s with key type %s as the source contains a null key.",
+            Map.class.getName(), keyType.getName());
+        Preconditions.checkNotNull(
+            value,
+            "Cannot get the value of a property of type %s with value type %s as the source contains a null value for key \"%s\".",
+            Map.class.getName(), valueType.getName(), key);
+
```

Comment:
:x: This validation should also be added to the collector for keys otherwise the public API will be inconsistent wrt validation. This is in `ValidatingValueCollector` and triggered when one uses `mapProperty.keySet()`

Reply:
Strictly speaking, the public API has not changed. A null pointer exception has been thrown earlier and it is thrown now. The only difference is that now there's a helpful error message.

But I'll try to add it in the other place as well.

Reply:
Any news here?